src/TileSlider.tsx

```diff
@@ -238,48 +241,66 @@ const TileSliderComponent = <T,>(
+    const totalDuration = DRAG_SNAPPING_DAMPING + extraDuration;
+    const blendDuration = Math.round(totalDuration / 2);
+    const swipeDirection = startVelocity < 0 ? 'left' : 'right';
+    const roundingFn = swipeDirection === 'left' ? Math.ceil : Math.floor;
+    const boundingFn = swipeDirection === 'left' ? Math.max : Math.min;
+    const boundingFnAlt = swipeDirection === 'left' ? Math.min : Math.max;
+
     cancelAnimationFrame(sliderDataRef.current.animationId);
 
-    // @todo set `toIndex` based on the velocity?
     setState((state) => ({ ...state, fromIndex: state.index, sliding: true }));
 
     const velocityDampening = () => {
       const currentTime = Date.now() - startTime;
-      const position = easeOutQuartic(currentTime, startVelocity, -startVelocity, totalDuration);
-      const currentIndex = calculateIndex();
+      const phase: 1 | 2 | 3 = currentTime >= totalDuration ? 3 : currentTime >= totalDuration - blendDuration ? 2 : 1;
+      const currentIndex = calculateIndex(roundingFn);
       const page = Math.floor(getCircularIndex(currentIndex, items.length) / tilesToShow);
+      const blendVelocity = 3;
+      const finalPosition = -(currentIndex * tileWidth);
 
-      sliderDataRef.current.position += position;
-
-      if (currentTime >= totalDuration - blendDuration) {
-        const targetPosition = -(currentIndex * tileWidth);
-        const change = targetPosition - sliderDataRef.current.position;
-        const currentBlendTime = Date.now() - (startTime + totalDuration - blendDuration);
+      let nextPosition = 0;
 
-        sliderDataRef.current.position = easeInOutCubic(currentBlendTime, sliderDataRef.current.position, change, blendDuration);
+      if (phase === 1) {
+        const movement = boundingFnAlt(blendVelocity, easeOutQuartic(currentTime, startVelocity, -startVelocity, totalDuration));
+        nextPosition = sliderDataRef.current.position + movement;
+      } else {
+        // Slowly snap to the target position
+        const blendStartTime = startTime + totalDuration - blendDuration;
+        const blendMs = Date.now() - blendStartTime;
+        const steps = Math.round(blendMs / 250);
+        const blendStep = blendVelocity / 12;
+        const blendMinimum = 0.8;
+        const currentBlendVelocity = Math.max(blendMinimum, blendVelocity - steps * blendStep);
+
+        if (swipeDirection === 'left') {
+          nextPosition = boundingFn(finalPosition, sliderDataRef.current.position - currentBlendVelocity);
+        } else {
+          nextPosition = boundingFn(finalPosition, sliderDataRef.current.position + currentBlendVelocity);
+        }
       }
 
+      sliderDataRef.current.position = nextPosition;
+
       // interrupt
       if (sliderDataRef.current.scrolling) return;
 
       frameRef.current.style.transform = `translateX(${sliderDataRef.current.position}px)`;
 
-      if (currentTime <= totalDuration) {
+      if (sliderDataRef.current.position !== finalPosition) {
```

Comment:
This is tricky because the changes are that the position will never reach the final position exactly.

Reply:
It should, theoretically, because `blendMinimum: 0.8` makes infinite momentum and then the `boundingFn` will set it exactly at the `finalPosition` at some point. But better to add safety indeed.

But maybe this PR should've been in draft actually. It was a proposal. :)